### DISCUSSION TITLE
GraphQL Actions API?

### DISCUSSION THREAD

**Discussion by jacek-jablonski** [community/community#24493] [Apps, API and Webhooks]
<p>Hi,</p>
<p>are Actions informations available through v4 GraphQL API? I couldn’t find anything related to GithubActions in docs.</p>
<p>Is this feature planned to be added? If yes, is there any timeline?</p>
<p>Cheers :slight_smile:</p>

**Comment by AndreaGriffiths11** [comment, marked as answer]
<p>Hi <a class="mention" href="/jacek-jablonski">@jacek-jablonski</a>,</p>
<p>Thanks for being here! As of now, the GitHub <a href="https://developer.github.com/v3/actions/" rel="nofollow noopener">Actions API</a> enables you to manage GitHub Actions only using the REST API. I’ll be sure to pass on your request for v4 support to the team for consideration.</p>

**Comment by bluedusk** [comment]
<p>Hi <a class="mention" href="/AndreaGriffiths11">@AndreaGriffiths11</a>    Any update on Actions Api support in V4 ?</p>

**Comment by bluedusk** [comment]
<p>Hi <a class="mention" href="/AndreaGriffiths11">@AndreaGriffiths11</a> Is there any plan to add Actions api to V4 ? If some new feature only available in rest api, how should I decide which version to use ?   Thanks</p>

**Comment by AndreaGriffiths11** [comment]
<p>Thanks for the ping <a class="mention" href="/bluedusk">@bluedusk</a> Regarding upcoming feature updates or new GitHub products, that’s not something we can speak to here. However, You can check our <a href="https://github.com/github/roadmap/projects/1">public roadmap</a>. The best place to submit feature requests and share more details about your use case is through our official product feedback <a href="https://support.github.com/contact/feedback">form</a> so that our product team can track your request.</p>
<p>I’m pinging the Actions PM on this post in case they might have any additional input. cc <a class="mention" href="/chrispat">@chrispat</a></p>

**Comment by chrispat** [comment]
<p>We currently do not have any backlog items to add GraphQL based API for Actions.</p>

**Comment by ramiroJCB** [comment]
Hello guys I just found this ticket from 2 years ago and I wanted to ask if there is any plan  for actually doing githubActions support for V4 ?

**Comment by sammcj** [comment]
Any update on using GraphQL for Actions?

**Reply by sammcj** [reply]
still waiting...

**Comment by KurtLehnardt** [comment]
Hi,

I'm currently tasked with finding a .zip artifact that is created during a GHA job. It's often buried among thousands of other artifacts and it's somewhat wasteful to iterate through pages and pages of API call results to find the specific artifact I'm looking for. 

I've used the GraphQL API to perform targeted requests on refs/tags artifacts, and would love to see the same functionality available for actions. 

Is there any roadmap for this functionality? 

Thanks!

**Comment by nathangriffiths-cdx** [comment]
Quite disappointing to find that the GraphQL API still doesn't support Gtihub Actions / workflow data after this amount of time.

**Comment by keer0322** [comment]
Is there any movement for this request ?S
Quite disappointed and surprised that  GitHub Action workflow info doesn't available through graphql considering GHA was released in 2018

**Comment by carlthome** [comment]
Surprised to see no GraphQL support for workflow runs. Any recent updates?

**Comment by tonyarnold** [comment]
Yeah, this is nuts — the disparity between GitHub's REST and GraphQL APIs is really disheartening. The quality issues with the REST API and its schemas led me to trying out the GraphQL API, only to find that entire parts of the API are missing entirely.

**Comment by kishaningithub** [comment]
In the current github documentation, i see workflow object mentioned (screenshot below) but i am unsure as to how to use it. Has anyone succeeded in using it?

Explorer Link: https://docs.github.com/en/graphql/overview/explorer

### Screenshot
<img width="337" alt="image" src="https://github.com/user-attachments/assets/1b4386a8-4afe-4a19-9269-f4b17b9826af" />

**Reply by futugyou** [reply]
[how-to-write-a-graphql-query-to-retrieve-all-the-workflows-runs-from-github](https://stackoverflow.com/questions/74054176/how-to-write-a-graphql-query-to-retrieve-all-the-workflows-runs-from-github)

I found this way, but it's not what I want.
I want to query the workflow directly in the repository.

**Reply by kishaningithub** [reply]
Same here @futugyou . Thanks for sharing though!

**Comment by AlexandreDecan** [comment]
They do not really care about making data available except when it comes to *your* data for *their* purposes. It's really unfortunate they don't understand how valuable these data can be for the community and, implicitly for them.

**Comment by sameagen-MW** [comment]
Hi @AndreaGriffiths11 and @chrispat ,

Are either of you aware of any current plans to add GraphQL support for GitHub Actions objects? It's been five years, so unfortunately the previously posted roadmap link is now broken. GraphQL support would be a lovely and impactful addition to the GHA API, and I'm curious whether it's worth it to keep holding out hope despite it seeming unlikely to happen at this point.

Thanks!

**Comment by Luiz-Monad** [comment]
Too bad, I guess I'm going to not only keep pooling the API, I will also have to keep paging it. Its a tremendous waste of resources, but oh well. Such a shame. Can you at least increase the amount of items per page to a 1000 ?
